### ISSUE TITLE
govc 0.24.0: device.pci.add: Incorrect DeviceID value

### ISSUE THREAD

**Issue by yet-another-aligator** [vmware/govmomi#2248]
**[Edit] Issue reproduced on a vSphere ESXi 6.7.0u3 host with same behavior.**

Hi,

I'm running govc v0.24.0 with a standalone vSphere ESXi 7.0u1 host and attempted to add a PCI passthru device to a VM by using the `device.pci.add` subcommand, which completed without error.

However, the VM couldn't be powered on : `The systemId does not match the current system or the deviceId and vendorId do not match the device currently at <bus:device.function>`.

The issue could be resolved manually by removing and adding again the PCI device from vSphere web client, as described [there](https://kb.vmware.com/s/article/2001216) but that's not compatible with my wish to automate my VI provisioning.

After a short investigation, I found that an incorrect DeviceID value was provided when adding PCI device to the VM :

---

**govc call**:

```
govc device.pci.add -vm VM-XXXX "0000:65:00.0"
```

---

**The real device ID (decimal value):**

```
govc host.esxcli -json hardware pci list | jq -e -r '.Values[] | select((.DeviceClass[] | select(. == "768")) and (.VendorID[] | select(. == "4318"))) | .DeviceID[0]'
```

**Output** : `7684`

---

**VMX data after adding PCI device with govc:**

```
pciPassthru0.id = "00000:101:00.0"
pciPassthru0.deviceId = "0x7684" <-- Incorrect value
pciPassthru0.vendorId = "0x10de"
pciPassthru0.systemId = "5fe9d4dc-68b5-75f6-e9ae-d45d64bbdaca"
pciPassthru0.present = "TRUE"
```

---

**VMX data after adding PCI device with vSphere Web Client:**

```
pciPassthru0.id = "00000:101:00.0"
pciPassthru0.deviceId = "0x1e04" <-- Correct value
pciPassthru0.vendorId = "0x10de"
pciPassthru0.systemId = "5fe9d4dc-68b5-75f6-e9ae-d45d64bbdaca"
pciPassthru0.present = "TRUE"
```

---

As you can read above, the incorrect DeviceID is actually DeviceID's decimal value but enclosed in an hexadecimal notation.

In `pci.device.add` source code,

https://github.com/vmware/govmomi/blob/1ec59a7c0002598c594fb58e03907d7ff0ca84b4/govc/device/pci/add.go#L133

the DeviceID is converted to string, as required by [VirtualPCIPassthroughDeviceBackingInfo object API](https://vdc-repo.vmware.com/vmwb-repository/dcr-public/3d076a12-29a2-4d17-9269-cb8150b5a37f/8b5969e2-1a66-4425-af17-feff6d6f705d/doc/vim.vm.device.VirtualPCIPassthrough.DeviceBackingInfo.html).

But is it possible that the string is expected in hexadecimal notation instead of decimal (which could explain the issue) ?

Thank you in advance for your help.

**Comment by yet-another-aligator** [comment]
Hi (again),

I just tested with :

```go
      Id: d.PciDevice.Id, DeviceId: fmt.Sprintf("%x", d.PciDevice.DeviceId),
```

instead of 

https://github.com/vmware/govmomi/blob/1ec59a7c0002598c594fb58e03907d7ff0ca84b4/govc/device/pci/add.go#L133

and I got the correct Device ID in VMX data (+ the VM powered on without error).

**Comment by yet-another-aligator** [comment]
https://github.com/vmware/govmomi/pull/2249

**Comment by dougm** [comment]
Fixed by #2249
